safe_relay_service/relay/services/circles_service.py

```diff
@@ -1,36 +1,59 @@
 from django.conf import settings
-from logging import getLogger
-
-from gnosis.eth.constants import NULL_ADDRESS
-from .transaction_service import TransactionServiceProvider
 
+from ethereum.utils import check_checksum
 
-logger = getLogger(__name__)
+from gnosis.eth import EthereumClient
+from gnosis.eth.constants import NULL_ADDRESS
 
 
 class CirclesService:
 
-    def __init__(self):
-        self.value = 0
+    def __init__(self, ethereum_client: EthereumClient, gas_price: int = 1):
+        """
+        :param ethereum_client:
+        :param gas_price: Gas price when paid in Circles token
+        """
+        self.ethereum_client = ethereum_client
+        self.gas_price = gas_price
 
-        # tx data from Circles Token contract signup method
-        self.data = ("0x519c6377000000000000000000000000000000000000000000000"
-            "0000000000000000020000000000000000000000000000000000000000"
-            "0000000000000000000000007436972636c65730000000000000000000"
-            "0000000000000000000000000000000")
+    def estimated_gas_price(self) -> int:
```

Comment:
I'm not sure this is an estimate? isn't this just returning a constant, set by us?

Reply:
I think we inherited this naming from gnosis but it might be good to change